pkg/chunk/grpc/index_client.go

```diff
@@ -0,0 +1,108 @@
+package grpc
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+
+	"github.com/cortexproject/cortex/pkg/chunk"
+	"github.com/cortexproject/cortex/pkg/chunk/util"
+)
+
+type writeBatch struct {
```

Comment:
Why do we have this. Why can't we use `WriteBatch` directly?

Reply:
I see Cassandra is maintaining the writeBatch implicitly within itself. So, I also added writeBatch within grpc client. I see dynamoDBstorageclient maintaining one within itself.

Reply:
Yes, but we have `grpc.WriteBatch` already. Why are we using another one? Specially when we're just copying this in `BatchWrite()`